projects/core/bindaas-core-impl/src/main/java/edu/emory/cci/bindaas/core/jwt/DefaultJWTManager.java

```diff
@@ -0,0 +1,128 @@
+package edu.emory.cci.bindaas.core.jwt;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.criterion.Restrictions;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTCreationException;
+import com.auth0.jwt.exceptions.JWTDecodeException;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.exceptions.SignatureVerificationException;
+import com.auth0.jwt.interfaces.Claim;
+import com.auth0.jwt.interfaces.DecodedJWT;
+
+import edu.emory.cci.bindaas.core.jwt.IJWTManager;
+import edu.emory.cci.bindaas.core.jwt.JWTManagerException;
+import edu.emory.cci.bindaas.core.jwt.JWTManagerException.Reason;
+import edu.emory.cci.bindaas.core.model.hibernate.HistoryLog;
+import edu.emory.cci.bindaas.core.model.hibernate.HistoryLog.ActivityType;
+import edu.emory.cci.bindaas.core.model.hibernate.UserRequest;
+import edu.emory.cci.bindaas.core.model.hibernate.UserRequest.Stage;
+import edu.emory.cci.bindaas.security.api.BindaasUser;
+
+import java.util.Date;
+import java.util.List;
+
+
+public class DefaultJWTManager implements IJWTManager {
+
+	private Log log = LogFactory.getLog(getClass());
+	private SessionFactory sessionFactory;
+	
+	public SessionFactory getSessionFactory() {
+		return sessionFactory;
+	}
+
+	public void setSessionFactory(SessionFactory sessionFactory) {
+		this.sessionFactory = sessionFactory;
+	}
+
+	private final static String secret = "fj32Jfv02Mq33g0f8ioDkw";
```

Comment:
Harcoded the value of secret here for easy debugging. Will generate randomly later.